src/System.Security.Principal.Windows/src/System/Security/Principal/WindowsImpersonationContext.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.ComponentModel;
+using System.Threading;
+using Microsoft.Win32.SafeHandles;
+
+namespace System.Security.Principal
+{
+    public class WindowsImpersonationContext : IDisposable
+    {
+        private readonly SafeAccessTokenHandle _previousUserToken = SafeAccessTokenHandle.InvalidHandle;
+        private readonly AsyncLocal<SafeAccessTokenHandle> _currentImpersonatedToken;
```

Comment:
Why is this needed? i.e. why can't this code just refer to s_currentImpersonatedToken (made internal instead of private)?

Reply:
can be done ok!

Reply:
done